src/ctap/token_state.rs

```diff
@@ -113,19 +111,18 @@ impl PinUvAuthTokenState {
     }
 
     /// Starts the timer for pinUvAuthToken usage.
-    pub fn begin_using_pin_uv_auth_token(&mut self, now: CtapInstant) {
+    pub fn begin_using_pin_uv_auth_token(&mut self) {
         self.user_verified = true;
-        self.usage_timer = TimedPermission::granted(now, INITIAL_USAGE_TIME_LIMIT);
+        self.usage_timer = Some(LibtockAlarmTimer::start(INITIAL_USAGE_TIME_LIMIT));
         self.in_use = true;
     }
 
     /// Updates the usage timer, and disables the pinUvAuthToken on timeout.
-    pub fn pin_uv_auth_token_usage_timer_observer(&mut self, now: CtapInstant) {
+    pub fn pin_uv_auth_token_usage_timer_observer(&mut self) {
         if !self.in_use {
             return;
         }
-        self.usage_timer = self.usage_timer.check_expiration(now);
-        if !self.usage_timer.is_granted(now) {
+        if self.usage_timer.is_none() || self.usage_timer.unwrap().has_elapsed().is_none() {
```

Comment:
This should also update the timer. We could actually provide a function for convenience:

```rust
/// Updates the expiration of a timer.
pub fn update_timer<T: Timer>(timer: &mut Option<T>) {
    *timer = core::mem::replace(timer, None).and_then(T::has_expired);
}
```